File: frontend/src/types/verifier.ts
/** Document verifier pending-queue types (E28; Journey J21). */

export interface PendingDocument {
  id: number
  tenant_id: number
  application_id: number
  checklist_item_template_id: number | null
  original_filename: string
  content_type: string
  size_bytes: number
  uploaded_by_user_id: number
  uploaded_at: string
  application_stage: string
  student_id: number
  university_id: number
  program_id: number
}

export interface PendingDocumentQueue {
  items: PendingDocument[]
  total: number
  limit: number
  offset: number
}

/** A document after a verifier decision (approve/reject) — E29/E30. */
export interface VerifiedDocument {
  id: number
  tenant_id: number
  application_id: number
  status: string
  original_filename: string
  verified_by_user_id: number | null
  verified_at: string | null
  rejection_reason: string | null
  approval_comment: string | null
}

File: backend/app/schemas/verifier.py
"""Schemas for the document verifier queue API (E28; Journey J21)."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.student_document import StudentDocumentStatus


class PendingDocumentItem(BaseModel):
    """Pending document queue entry, including application context."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    tenant_id: int
    application_id: int
    checklist_item_template_id: int | None
    status: StudentDocumentStatus
    original_filename: str
    content_type: str
    size_bytes: int
    uploaded_by_user_id: int
    uploaded_at: datetime
    application_stage: str
    student_id: int
    university_id: int
    program_id: int


class PendingDocumentQueueResponse(BaseModel):
    """Paginated response for the document verifier queue."""

    items: list[PendingDocumentItem]
    total: int
    limit: int
    offset: int
